spec/rspec/mocks/matchers/have_received_spec.rb

```diff
@@ -1,6 +1,26 @@
 module RSpec
   module Mocks
-    RSpec.describe Matchers::HaveReceived do
+    # This shared example group is highly unusual as it is used to test how
+    # `have_received` works in two situations:
+    #
+    # * With rspec-mocks as a standalone library.
+    # * Together with rspec-expectations.
+    #
+    # To simulate the former, we've had to hack things a bit. Special care must be taken:
+    #
+    # * Only define examples with `it`, (not `fit`, `xit`, `specify`, etc). We redefine
+    #   `it` below to make it support our needs here but that definition isn't applied to
+    #   the other forms.
+    # * All normal expectations must use `_expect`, not `expect`. Expectations
+    #   for `have_received` should use `expect`.
+    RSpec.shared_examples_for Matchers::HaveReceived do
+      # Make rspec-expectations' `expect` available via an alias so that when
+      # this group is included below in a context that simulates rspec-expectations
+      # not being loaded by using rspec-mocks' `expect` instead of rspec-expectations'
+      # `expect`, we still have a way to use the expectations one for normal expectations.
+      # In this group, all normal expectations should use `_expect` instead of `expect`.
+      alias _expect expect
```

Comment:
Would making this name longer (e.g. `rspec_expectations_expect`) or changing the other one to be "mocks_expect" add clarity here?